packages/core-sdk/resolvers/token-resolvers/resolvers/compound/src/query/__tests__/e2e/integration.spec.ts

```diff
@@ -0,0 +1,136 @@
+    const api = await buildAndDeployApi(apiPath, testEnvState.ipfs, testEnvState.ensAddress);
+    protocolEnsUri = `ens/testnet/${api.ensDomain}`;
+
+    // deploy token defiwrapper
+    const tokenApiPath: string = path.join(apiPath, "../../../../", "token");
+    const tokenApi = await buildAndDeployApi(
+      tokenApiPath,
+      testEnvState.ipfs,
+      testEnvState.ensAddress,
+    );
+    tokenEnsUri = `ens/testnet/${tokenApi.ensDomain}`;
+  });
+
+  afterAll(async () => {
+    await stopTestEnvironment();
+  });
+
+  describe("isValidProtocolToken", () => {
+    test("compound_v1 cWBTC", async () => {
+      const result = await isValidProtocolToken(v1_cWBTC, "compound_v1", protocolEnsUri, client);
+      expect(result.error).toBeFalsy();
+      expect(result.data).not.toBeUndefined();
+      expect(result.data).toBe(true);
+    });
+
+    test("compound_v1 invalid protocol token", async () => {
+      const result = await isValidProtocolToken(WBTC, "compound_v1", protocolEnsUri, client);
+      expect(result.error).toBeFalsy();
+      expect(result.data).not.toBeUndefined();
+      expect(result.data).toBe(false);
+    });
+  });
+
+  describe("getTokenComponents", () => {
+    test("compound_v1 cWBTC", async () => {
+      const result = await getTokenComponents(
+        v1_cWBTC,
+        "compound_v1",
+        tokenEnsUri,
+        protocolEnsUri,
+        client,
+      );
+
+      expect(result.error).toBeFalsy();
+      expect(result.data).toBeTruthy();
+      expect(result.data).toMatchObject({
+        rate: "1",
+        unresolvedComponents: 0,
+        tokenAddress: v1_cWBTC,
+        components: [
+          {
+            tokenAddress: WBTC,
+            components: [],
+            unresolvedComponents: 0,
+          },
+        ],
+      });
+      const tokenComponent = result.data as TokenComponent;
+      let sum = 0;
+      tokenComponent.components.forEach((x: TokenComponent) => {
+        sum += +x.rate;
+      });
+      expect(sum).toBeGreaterThan(0);
+      expect(sum).toBeLessThan(1);
+    });
+
+    test("compound_v1 cETH", async () => {
+      const result = await getTokenComponents(
+        v1_cETH,
+        "compound_v1",
+        tokenEnsUri,
+        protocolEnsUri,
+        client,
+      );
+
+      expect(result.error).toBeFalsy();
+      expect(result.data).toBeTruthy();
+      expect(result.data).toMatchObject({
+        rate: "1",
+        unresolvedComponents: 0,
+        tokenAddress: v1_cETH,
+        components: [
+          {
+            tokenAddress: ETH,
+            components: [],
+            unresolvedComponents: 0,
+          },
+        ],
+      });
+      const tokenComponent = result.data as TokenComponent;
+      let sum = 0;
+      tokenComponent.components.forEach((x: TokenComponent) => {
+        sum += +x.rate;
+      });
+      expect(sum).toBeGreaterThan(0);
+      expect(sum).toBeLessThan(1);
```

Comment:
This will always be around `0.02` since Geometrical Progression formed by the interest rate accrual increases very slowly and bound by upper limit even in case of (n -> inf where n = no. of blocks)

Reply:
I adjusted the expected test results. Thanks!